### PR TITLE
[Shortcircuit API] Add detailedReport in parameters

### PR DESCRIPTION
**Pull request by colinepiloquet**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

No parameter

**What is the new behavior (if this is a feature change)?**
Adds a parameters indicating if the functional logs (from reporter) should be detailed or aggregated

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitParameters.java [hunk 1]
```diff
@@ -82,7 +84,8 @@ public static ShortCircuitParameters load(PlatformConfig platformConfig) {
                         .setWithVSCConverterStations(config.getBooleanProperty("with-vsc-converter-stations", DEFAULT_WITH_VSC_CONVERTER_STATIONS))
                         .setWithNeutralPosition(config.getBooleanProperty("with-neutral-position", DEFAULT_WITH_NEUTRAL_POSITION))
                         .setInitialVoltageProfileMode(config.getEnumProperty("initial-voltage-profile-mode", InitialVoltageProfileMode.class, DEFAULT_INITIAL_VOLTAGE_PROFILE_MODE))
-                        .setVoltageRanges(getVoltageRangesFromConfig(config)));
+                        .setVoltageRanges(getVoltageRangesFromConfig(config))
+                        .setDetailedReport(config.getBooleanProperty("detailed-report", DEFAULT_DETAILED_REPORT)));
```
Comment (olperr1) [inline shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitParameters.java:88]: This parameter can be tested in adding:

- in `src/test/resources/config.yml`:
```yml
detailed-report: false
```
- in `ShortCircuitParametersTest.testConfigLoader()`:
```java
assertFalse(parameters.isDetailedReport());
```

File: shortcircuit-api/src/test/java/com/powsybl/shortcircuit/ShortCircuitParametersTest.java [hunk 2]
```diff
@@ -278,7 +278,35 @@ void readVersion12() {
...
+        assertTrue(parameters.isWithNeutralPosition());
+        assertEquals(InitialVoltageProfileMode.CONFIGURED, parameters.getInitialVoltageProfileMode());
+        List<VoltageRange> voltageRanges = parameters.getVoltageRanges();
+        assertEquals(3, voltageRanges.size());
+        assertEquals(1.05, voltageRanges.get(0).getRangeCoefficient());
+        assertEquals(Range.between(380., 410.), voltageRanges.get(0).getRange());
+        assertEquals(1.1, voltageRanges.get(1).getRangeCoefficient());
+        assertEquals(Range.between(0., 225.), voltageRanges.get(1).getRange());
+        assertEquals(1.09, voltageRanges.get(2).getRangeCoefficient());
+        assertEquals(Range.between(230., 375.), voltageRanges.get(2).getRange());
```
Comment (olperr1) [inline shortcircuit-api/src/test/java/com/powsybl/shortcircuit/ShortCircuitParametersTest.java:294]: The default value can be tested, for instance here:
```suggestion
        assertEquals(Range.between(230., 375.), voltageRanges.get(2).getRange());
        assertTrue(parameters.isDetailedReport());
```

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitParameters.java (hunk 1 above)
Comment (colinepiloquet) [reply shortcircuit-api/src/main/java/com/powsybl/shortcircuit/ShortCircuitParameters.java:88]: You are right, I will add that right now, thank you

File: shortcircuit-api/src/test/java/com/powsybl/shortcircuit/ShortCircuitParametersTest.java (hunk 2 above)
Comment (colinepiloquet) [reply shortcircuit-api/src/test/java/com/powsybl/shortcircuit/ShortCircuitParametersTest.java:294]: Actually this is just an improvement of the test of version 1.2 that does not have this new parameter. The new parameters introduced in this version were not introduced so I added them, like that the test make more sense, testing all the new parameters. 
To test `detailedReport`, I created a test that is right below

Comment (olperr1) [reply shortcircuit-api/src/test/java/com/powsybl/shortcircuit/ShortCircuitParametersTest.java:295]: I thought that the default value of the parameter wasn't covered by any test. But in fact, `writeExtension()` will fail if the default value is changed. 
=> No need for another test.

**Review by olperr1 — APPROVED** [review]

**Pull request merged**